sdk/python/feast/errors.py

```diff
@@ -477,6 +477,16 @@ def __init__(self, name, project=None):
             super().__init__(f"Permission {name} does not exist")
 
 
+class ProjectNotFoundException(Exception):
```

Comment:
Why are both of these necessary? what's the difference. btw, All Feast exceptoins should extend `FeastError`.

Reply:
@tokoko I noticed the same while doing the latest refactory but I didn't want to extend the scope of changes too much. Looks like other duplications are there like `PermissionObjectNotFoundException` and `DataSourceObjectNotFoundException`. Can we remove all of them (in a dedicated ticket maybe)?

Reply:
I'll extend the FeastError. 
Reasons for having two exceptions: Just followed the current convention. 
ProjectNotFoundException --> Exception thrown when the DELETE calls are made and project is not found. 
ProjectObjectNotFoundException --> Exception thrown when GET calls are made and project is not found. _get_object method passes two args (name, project) when object is not found. I don't want to add another if else logic to filter out project.

Reply:
@dmartinol yeah, this just looks weird. Probably best to defer removal until after this is merged, though.. just to avoid a messy rebase

Reply:
in case, remember an item in the next Release Notes to remind clients to replace any try/except that may have used those duplicate errors.